src/cardano/CardanoEngine.ts

```diff
@@ -0,0 +1,556 @@
+    }
+
+    const {
+      min_fee_a: minFeeA,
+      min_fee_b: minFeeB,
+      max_tx_size: maxTxSize,
+      key_deposit: keyDeposit,
+      pool_deposit: poolDeposit,
+      max_val_size: maxValSize,
+      coins_per_utxo_size: coinsPerUtxoSize
+    } = this.tools.epochParams
+
+    const txBuilderConfig = Cardano.TransactionBuilderConfigBuilder.new()
+      .fee_algo(
+        Cardano.LinearFee.new(
+          Cardano.BigNum.from_str(minFeeA.toString()),
+          Cardano.BigNum.from_str(minFeeB.toString())
+        )
+      )
+      .coins_per_utxo_byte(Cardano.BigNum.from_str(coinsPerUtxoSize))
+      .max_value_size(maxValSize)
+      .max_tx_size(maxTxSize)
+      .pool_deposit(Cardano.BigNum.from_str(poolDeposit))
+      .key_deposit(Cardano.BigNum.from_str(keyDeposit))
+      .build()
+    const txBuilder = Cardano.TransactionBuilder.new(txBuilderConfig)
+
+    const outputAddr = Cardano.Address.from_bech32(outputAddress)
+    const changeAddr = Cardano.Address.from_bech32(
+      this.walletInfo.keys.bech32Address
+    )
+
+    txBuilder.set_ttl_bignum(
+      Cardano.BigNum.from_str(this.slot.toString()).checked_add(
+        Cardano.BigNum.from_str('7200')
+      )
+    )
+
+    // Add output to the tx
+    txBuilder.add_output(
+      Cardano.TransactionOutput.new(
+        outputAddr,
+        Cardano.Value.new(Cardano.BigNum.from_str(outputAmount))
+      )
+    )
+
+    // Filter out multi asset utxo to keep this simple
+    const lovelaceUtxos = this.utxos.filter(u => u.value !== '0')
+
+    // Create TransactionUnspentOutputs from utxos fetched
+    const unspentOutputs = Cardano.TransactionUnspentOutputs.new()
+    for (const utxo of lovelaceUtxos) {
+      const inputValue = Cardano.Value.new(Cardano.BigNum.from_str(utxo.value))
+      const input = Cardano.TransactionInput.new(
+        Cardano.TransactionHash.from_bytes(Buffer.from(utxo.tx_hash, 'hex')),
+        utxo.tx_index
+      )
+      const output = Cardano.TransactionOutput.new(changeAddr, inputValue)
+      unspentOutputs.add(Cardano.TransactionUnspentOutput.new(input, output))
+    }
+    txBuilder.add_inputs_from(
+      unspentOutputs,
+      Cardano.CoinSelectionStrategyCIP2.LargestFirst
+    )
+
+    // Adds a change output if there are more ADA in utxo than we need for the transaction,
+    // these coins will be returned to change address
+    txBuilder.add_change_if_needed(changeAddr)
+
+    // Build transaction
+    const txBody = txBuilder.build()
+
+    switch (this.networkInfo.networkId) {
+      case 1:
+        txBody.set_network_id(Cardano.NetworkId.mainnet())
+        break
+      case 0:
+        txBody.set_network_id(Cardano.NetworkId.testnet())
+        break
+      default:
+        throw new Error('Unknown networkId')
+    }
+
+    return txBody
+  }
+
+  async makeSpend(edgeSpendInfoIn: EdgeSpendInfo): Promise<EdgeTransaction> {
+    edgeSpendInfoIn = upgradeMemos(edgeSpendInfoIn, this.currencyInfo)
```

Comment:
Maybe we could drop the legacy memo support for new coins going forward? Dropping it for Ethereum or whatever would be a breaking change, but for new stuff, there's no legacy support to break.